temporal/fx.go

```diff
@@ -549,13 +548,7 @@ func WorkerServiceProvider(
 
 	if _, ok := params.ServiceNames[serviceName]; !ok {
 		params.Logger.Info("Service is not requested, skipping initialization.", tag.Service(serviceName))
-		return ServicesGroupOut{
-			Services: &ServicesMetadata{
-				App:           fx.New(fx.NopLogger),
-				ServiceName:   serviceName,
-				ServiceStopFn: func() {},
-			},
-		}, nil
```

Comment:
This is a great change, but can we move it to a separate PR?

Reply:
I split out the Collection one (already in a separate commit) but I'd rather not spend time on splitting the rest, they're fairly intertwined